CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+## How to Contribute
+
+The OCS-Operator project is under the [Apache 2.0 license](LICENSE). We accept
+contributions via GitHub pull requests. This document outlines how to
+contribute to the project.
+
+## Contribution Flow
+
+Developers must follow these steps to make a change:
+
+1. Fork the `openshift/ocs-operator` repository on GitHub.
+2. Create a branch from the master branch, or from a versioned branch (such
+   as `release-4.2`) if you are proposing a backport.
+3. Make changes.
+4. Create tests as needed and ensure that all tests pass.
+5. Ensure your commit messages include sign-off.
+6. Push your changes to a branch in your fork of the repository.
+7. Submit a pull request to the `openshift/ocs-operator` repository.
+8. Work with the community to make any necessary changes through the code
+   review process (effectively repeating steps 3-8 as needed).
+
+## Developer Environment Installation
+
+Instructions to create a dev environment for OCS-operator can be found in the
+[main project documentation](./README.md#Installation of development builds).
+
+## Commit Requirements
+
+In addition to passing [functional tests](./README.md#Functional Tests), all
+commits to OCS-operator must follow the guidelines in this section. You can
+verify your changes meet most of these standards by running the following
+command in the repository's root directory:
+
+```
+make ocs-operator-ci
+```
+
```

Comment:
Could you add some more paragraphs about the expected commit structure? e.g. we want small, logically self-contained commits, each commit should pass tests, etc.

Reply:
Sure, absolutely. Since those will likely be a bit more controversial, I thought to include those in separate PRs, but I am happy to have those discussions on the first round instead.